### ISSUE TITLE
Kodi not quitting when launching Steam

### ISSUE THREAD

**Issue by twig** [teeedubb/teeedubb-xbmc-repo#5]
Hi there,

I recently upgraded to Kodi and found the addon to not exit Kodi after launching Steam BPM.

When I activate the addon, this is the debug log:

    14:41:47 T:5220  NOTICE: -->Python Interpreter Initialized<--
    14:41:47 T:5220  NOTICE: script.steam.launcher: ****Running Steam-Launcher v3.1.1....
    14:41:47 T:5220  NOTICE: script.steam.launcher: running on osAndroid, osOsx, osLinux, osWin: 0 0 0 1
    14:41:47 T:5220  NOTICE: script.steam.launcher: System text encoding in use: mbcs
    14:41:47 T:5220  NOTICE: script.steam.launcher: usr scripts are set to not be checked for updates, skipping version check
    14:41:47 T:5220  NOTICE: script.steam.launcher: skipping deleting userdata scripts, option disabled: delUserScriptSett = false
    14:41:47 T:5220  NOTICE: script.steam.launcher: script file already exists, skipping copy to userdata: C:\Users\twig\AppData\Roaming\Kodi\userdata\addon_data\script.steam.launcher\scripts\SteamLauncher-AHK.ahk
    14:41:47 T:5220  NOTICE: script.steam.launcher: script file already exists, skipping copy to userdata: C:\Users\twig\AppData\Roaming\Kodi\userdata\addon_data\script.steam.launcher\scripts\SteamLauncher-AHK.exe
    14:41:47 T:5220  NOTICE: script.steam.launcher: skipping program file check, option disabled: filePathCheck = false
    14:41:47 T:5220  NOTICE: script.steam.launcher: pre steam script: false
    14:41:47 T:5220  NOTICE: script.steam.launcher: post steam script: false
    14:41:47 T:5220  NOTICE: script.steam.launcher: quit setting selected: 0
    14:41:47 T:5220  NOTICE: false
    14:41:47 T:5220  NOTICE: script.steam.launcher: attempting to launch: "C:\Users\twig\AppData\Roaming\Kodi\userdata\addon_data\script.steam.launcher\scripts\SteamLauncher-AHK.exe" "E:\Games\Steam\Steam.exe" "C:\Program Files (x86)\Kodi\Kodi.exe" "0" "false" "false" "false"
    14:41:47 T:5220  NOTICE: "C:\Users\twig\AppData\Roaming\Kodi\userdata\addon_data\script.steam.launcher\scripts\SteamLauncher-AHK.exe" "E:\Games\Steam\Steam.exe" "C:\Program Files (x86)\Kodi\Kodi.exe" "0" "false" "false" "false"
    14:41:47 T:5220  NOTICE: script.steam.launcher: busy dialog started
    14:41:52 T:5220  NOTICE: script.steam.launcher: busy dialog stopped after: 5 seconds

When I press alt+tab, I can still see Kodi running in the background (and dimming the screen when there hasn't been any mouse/keyboard input when I play with control pad).

![steam kodi alt tab](https://cloud.githubusercontent.com/assets/59689/8028345/5faae460-0dee-11e5-8d9c-066b1527210f.png)

I tried adding MsgBox to these lines but it didn't seem to ever show.

    WinWait, ahk_group SteamBPM
    if %3%
    {
    	MsgBox Case A
    	WinMinimize, Kodi ahk_class Kodi
    }
    else
    {
    	MsgBox Case B
    	Run, %comspec% /c taskkill /im Kodi.exe,,Hide
    	Run, %comspec% /c timeout /t 1 && tasklist /nh /fi "imagename eq Kodi.exe" | find /i "Kodi.exe" >nul && (taskkill /f /im Kodi.exe),,Hide
    }

If you need any more information, please let me know.

**Comment by twig** [comment]
Forgot to mention;

* Windows 7 x64
* Steam Package version 1433441724
* Kodi 14.2

**Comment by Scondo** [comment]
You need to recompile AHK scripts after changes.

**Comment by twig** [comment]
Hey @Scondo,

1. I'm only messing with the AHK file in the Kodi addons folder because I have AutoHotKey installed. Does the recompilation thing mean I have to turn it into an EXE?

2. I will check once I get home.

**Comment by Scondo** [comment]
Yes, you have to turn .ahk to .exe and replace "SteamLauncher-AHK.exe" file in "addon_data" folder.

**Comment by teeedubb** [comment]
Some other people have reported the same but I cant reproduce it on any of my systems so this is a tough one to track down.

Do as Scondo suggested, the ahk script + exe you need to edit + compile are located in C:\Users\twig\AppData\Roaming\Kodi\userdata\addon_data\script.steam.launcher\scripts\

Did the addon prompt you to upgrade your scripts after updating to kodi? If not select the option 'delete and update userdata scripts on next run' this will update the bundled scripts.

Also are you running kodi as admin? How are you launching kodi? With kodi running, does runnning

"C:\Users\twig\AppData\Roaming\Kodi\userdata\addon_data\script.steam.launcher\scripts\SteamLauncher-AHK.exe" "E:\Games\Steam\Steam.exe" "C:\Program Files (x86)\Kodi\Kodi.exe" "0" "false" "false" "false"

via command quit kodi before running steam?

**Comment by twig** [comment]
Thanks guys.

I've compiled the script and added a debug MsgBox at the start

    MsgBox 1 = %1% \n2 = %2% \n3 = %3% \n4 = %4% \n5 = %5% \n6 = %6%

And got this.
![steam launcher debug](https://cloud.githubusercontent.com/assets/59689/8056827/600d4ca2-0eef-11e5-9288-b409fd4ab4da.png)

After I recompiled the AHK to exe and replaced the files in `%APPDATA%\Kodi\userdata\addon_data\script.steam.launcher\scripts`, it started exiting correctly again.

Probably something that didn't update correctly when I upgraded from XBMC 12 to Kodi 14.
